src/api/store/community.py

```diff
@@ -1,16 +1,206 @@
-from database.models import Community, CommunityMember, UserProfile, Action, Event, Graph, Media, ActivityLog, AboutUsPageSettings, ActionsPageSettings, ContactUsPageSettings, DonatePageSettings, HomePageSettings, ImpactPageSettings, TeamsPageSettings, Goal, CommunityAdminGroup
+from database.models import Community, CommunityMember, UserProfile, Action, Event, Graph, Media, ActivityLog, AboutUsPageSettings, ActionsPageSettings, ContactUsPageSettings, DonatePageSettings, HomePageSettings, ImpactPageSettings, TeamsPageSettings, Goal, CommunityAdminGroup, Location, RealEstateUnit
 from _main_.utils.massenergize_errors import MassEnergizeAPIError, InvalidResourceError, ServerError, CustomMassenergizeError
 from _main_.utils.massenergize_response import MassenergizeResponse
 from _main_.utils.context import Context
 from django.db.models import Q
-from .utils import get_community_or_die, get_user_or_die, get_new_title
-import random 
+from .utils import get_community_or_die, get_user_or_die, get_new_title, is_reu_in_community
+import random
+import zipcodes
 from sentry_sdk import capture_message, capture_exception
 
 class CommunityStore:
   def __init__(self):
     self.name = "Community Store/DB"
 
+  # QUESTION FOR SAM:
+  # These helper routines don't make use of self, should they be member functions as currently implemented, 
+  # or outside the CommunityStore class?
+
+  def _check_geography_unique(self, community, geography_type, loc):
+    # ensure that this zip code is not part of another geographic zip code based community
+    check_communities = Community.objects.filter(is_geographically_focused=True, geography_type=geography_type, is_deleted=False).prefetch_related('location_set')
+    for check_community in check_communities:
```

Comment:
yeaah this is probably not wrong, but very expensive

Reply:
No it isn't wrong, it is correct and I've checked it.  There might be ways of speeding up,  but this routine is only used when a super admin from the admin portal creates or modifies a community.  It runs quite quickly so I'm not at all worried about the computation time.   Unless you mean expensive in $$ and then I'd like you to explain why.